### PR TITLE
v1.14 Backports 2024-07-23

### PR DESCRIPTION
**Pull request by joamaki**
* [x] #33845 (@giorio94)
 * [ ] #33823 (@tommyp1ckles)
 * [x] #33819 (@giorio94) :warning: resolved conflicts
 * [x] #33887 (@gandro)
 * [x] #33934 (@giorio94) :warning: resolved conflicts

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 33845 33823 33819 33887 33934
```

### CODE REVIEW

File: .github/workflows/conformance-aws-cni.yaml [hunk 1]
```diff
@@ -369,7 +372,17 @@ jobs:
...
           status: ${{ needs.installation-and-connectivity.result }}
+      - name: Set final commit status
+        if: ${{ needs.installation-and-connectivity.result == 'skipped' }}
+        uses: myrotvorets/set-commit-status-action@3730c0a348a2ace3c110851bed53331bc6406e9f # v2.0.1
+        with:
+          sha: ${{ inputs.SHA || github.sha }}
+          status: ${{ github.event_name != 'schedule' && 'success' || 'failure' }}
+          description: 'Skipped'
+
+
```
Comment (giorio94) [inline .github/workflows/conformance-aws-cni.yaml]: Nit. It could make sense to drop these extra new lines, as they could trigger conflicts in the future.

**Review by giorio94 — APPROVED** [review]
My commits look good, thanks!

**Comment by joamaki** [comment]
/test-backport-1.14

**Review by viktor-kurchenko — APPROVED** [review]

**Review by squeed — APPROVED** [review]

**Comment by joamaki** [comment]
/test-backport-1.14

**Comment by squeed** [comment]
gandro is on vacation, approved for him.

**Review by tommyp1ckles — APPROVED** [review]

**Pull request merged**